src/consensus/aft/raft.h

```diff
@@ -102,6 +102,17 @@ namespace aft
     bool is_execution_pending = false;
     std::list<std::unique_ptr<AbstractMsgCallback>> pending_executions;
 
+    // When this node receives append entries from a new primary, it may need to
+    // roll back a committable but uncommitted suffix it holds. The
+    // new primary dictates the index where this suffix begins, which
+    // following the Raft election rules must be at least as high as the highest
+    // commit index reported by the previous primary. The window in which this
+    // rollback could be accepted is minimised to avoid unnecessary
+    // retransmissions - this node only executes this rollback instruction on
+    // the first append entries after it became a follower. As with any append
+    // entries, the initial index will not advance until this node acks.
+    bool is_new_follower = false;
```

Comment:
This is an accurate state flag name, but I wonder if a condition-based name like `waiting_for_initial_view_seqno` or `pending_end_of_view_rollback` would be more descriptive. Given the extensive comment, this is nitpicking at best.